nx/lib/nx.ex

```diff
@@ -6365,28 +6365,101 @@ defmodule Nx do
         f32[3]
         [2.0, 1.0, 2.0]
       >
+
+  ## Vectorized tensors
+
+  Vectorized and non-vectorized tensors can be mixed-and-matched on all three inputs.
+
+      iex> pred = Nx.tensor([[0, 1, 0], [1, 1, 0]]) |> Nx.vectorize(:x)
+      iex> on_true = 1
+      iex> on_false = Nx.tensor([2, 3]) |> Nx.vectorize(:y)
+      iex> Nx.select(pred, on_true, on_false)
+      #Nx.Tensor<
+        vectorized[x: 2][y: 2]
+        s64[3]
+        [
+          [
+            [2, 1, 2],
+            [3, 1, 3]
+          ],
+          [
+            [1, 1, 2],
+            [1, 1, 3]
+          ]
+        ]
+      >
+
+    In the next example, notice that even though the `pred` input
+    is scalar, because we're dealing with vectorize inputs, some
+    broadcasting still occurs.
+
+      iex> pred = 1
+      iex> on_true = Nx.tensor([1, 2, 3]) |> Nx.vectorize(:x)
+      iex> on_false = Nx.tensor([4, 5]) |> Nx.vectorize(:y)
+      iex> Nx.select(pred, on_true, on_false)
+      #Nx.Tensor<
+        vectorized[x: 3][y: 2]
+        s64
+        [
+          [1, 1],
+          [2, 2],
+          [3, 3]
+        ]
+      >
+
+  Finally, broadcasting will also occur if more than one input share
+  the same vectorized axes, but one of them presents size 1
+
+      iex> pred = Nx.tensor([1, 0, 0]) |> Nx.vectorize(:x)
+      iex> on_true = Nx.tensor([[2]]) |> Nx.vectorize(:x) |> Nx.vectorize(:y)
+      iex> on_false = Nx.tensor([3, 4]) |> Nx.vectorize(:y)
+      iex> Nx.select(pred, on_true, on_false)
+      #Nx.Tensor<
+        vectorized[x: 3][y: 2]
+        s64
+        [
+          [2, 2],
+          [3, 4],
+          [3, 4]
+        ]
+      >
   """
   @doc type: :element
   def select(pred, on_true, on_false) do
-    %T{shape: pred_shape, names: pred_names} = pred = to_tensor(pred)
-    %T{shape: true_shape, names: true_names} = on_true = to_tensor(on_true)
-    %T{shape: false_shape, names: false_names} = on_false = to_tensor(on_false)
+    [pred, on_true, on_false] = broadcast_vectors([pred, on_true, on_false])
 
-    Nx.Shared.raise_vectorized_not_implemented_yet(pred, __ENV__.function)
-    Nx.Shared.raise_vectorized_not_implemented_yet(on_true, __ENV__.function)
-    Nx.Shared.raise_vectorized_not_implemented_yet(on_false, __ENV__.function)
+    %T{vectorized_axes: vectorized_axes, shape: pred_shape, names: pred_names} = pred
+    %T{shape: true_shape, names: true_names} = on_true
+    %T{shape: false_shape, names: false_names} = on_false
 
     output_type = binary_type(on_true, on_false)
 
     {output_shape, output_names} =
-      case pred_shape do
-        {} ->
+      case {vectorized_axes, pred_shape} do
+        {[], {}} ->
           Nx.Shape.binary_broadcast(true_shape, true_names, false_shape, false_names)
 
         _ ->
-          {pred_shape, pred_names}
```

Comment:
Should we still match on `{[], _}` and return `{pred_shape, pred_names}`?

Reply:
Makes sense. Avoids some work :)